addons/account/models/account_invoiceable_order.py

```diff
@@ -0,0 +1,226 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import Command, _, api, fields, models
+from odoo.tools import float_is_zero
+
+
+class InvoiceableOrderMixin(models.AbstractModel):
+    _name = 'account.order'
+    _description = 'Enables downpayment support for models that can have down payments on them'
+
+    amount_to_invoice = fields.Monetary("Amount to invoice", store=True, compute='_compute_amount_to_invoice')
+    amount_invoiced = fields.Monetary("Already invoiced", compute='_compute_amount_invoiced')
+
+    # To override
+    company_id = fields.Many2one('res.company')
+    currency_id = fields.Many2one('res.currency')
+    fiscal_position_id = fields.Many2one('account.fiscal.position', string='Fiscal Position')
+    partner_id = fields.Many2one('res.partner')
+
+    amount_total = fields.Monetary()
+
+    invoice_ids = fields.Many2many('account.move')
+    order_line = fields.One2many('account.order.line')
```

Comment:
Yes it was some kind of "abstract declaration" because it felt weird to use an undeclared field in the abstract class methods, like magically assuming that all subclasses will just have a field with that name pointing to some subclass of the abstract class. But as it is useless/not necessary would you prefer me to remove it here and just keep it on the subclasses?

Reply:
Hmmmm :thinking: Not sure. I don't know if we had other cases like that in the past. I suggest you ask the ORM team what they think about it. Maybe there's just a function we can override to check that the proper fields are declared on the subclasses ? (I doubt it)